src/components/Tooltip/Tooltip.tsx

```diff
@@ -130,58 +149,80 @@ const Tooltip = ({
     }
 
     enabledEvents.forEach(({ event, listener }) => {
-      elementReference?.addEventListener(event, listener)
+      elementRefs.forEach((ref) => {
+        ref.current?.addEventListener(event, listener)
+      })
     })
 
     return () => {
       enabledEvents.forEach(({ event, listener }) => {
-        elementReference?.removeEventListener(event, listener)
+        elementRefs.forEach((ref) => {
+          ref.current?.removeEventListener(event, listener)
+        })
       })
     }
-  }, [anchorId, events, delayHide, delayShow])
+  }, [anchorRefs, anchorId, events, delayHide, delayShow])
 
   useEffect(() => {
-    const elementReference = document.querySelector(`[id='${anchorId}']`)
-
-    computeToolTipPosition({
+    let elementReference = activeAnchor.current
+    if (anchorId) {
+      // `anchorId` element takes precedence
+      elementReference = document.querySelector(`[id='${anchorId}']`) as HTMLElement
+    }
+    setCalculatingPosition(true)
+    let mounted = true
+    computeTooltipPosition({
       place,
       offset,
       elementReference,
-      tooltipReference: tooltipRef.current as HTMLElement,
-      tooltipArrowReference: tooltipArrowRef.current as HTMLElement,
+      tooltipReference: tooltipRef.current,
+      tooltipArrowReference: tooltipArrowRef.current,
       strategy: positionStrategy,
     }).then((computedStylesData) => {
+      if (!mounted) {
+        // invalidate computed positions after remount
+        return
+      }
+      setCalculatingPosition(false)
       if (Object.keys(computedStylesData.tooltipStyles).length) {
         setInlineStyles(computedStylesData.tooltipStyles)
       }
-
       if (Object.keys(computedStylesData.tooltipArrowStyles).length) {
         setInlineArrowStyles(computedStylesData.tooltipArrowStyles)
       }
     })
+    return () => {
+      mounted = false
+    }
+  }, [show, isOpen, anchorId, activeAnchor, place, offset, positionStrategy])
 
+  useEffect(() => {
     return () => {
-      tooltipShowDelayTimerRef.current = undefined
-      tooltipHideDelayTimerRef.current = undefined
+      if (tooltipShowDelayTimerRef.current) {
+        clearTimeout(tooltipShowDelayTimerRef.current)
+      }
+      if (tooltipHideDelayTimerRef.current) {
+        clearTimeout(tooltipHideDelayTimerRef.current)
+      }
     }
-  }, [show, isOpen, anchorId])
+  }, [])
 
   return (
     <WrapperElement
       id={id}
       role="tooltip"
       className={classNames(styles['tooltip'], styles[variant], className, {
-        [styles['show']]: isOpen || show,
+        [styles['show']]: !calculatingPosition && (isOpen || show),
```

Comment:
The current lifecycle of the tooltip with multiple anchors:

1. New active anchor is set
2. Position calculation (in `Tooltip`) and data attributes observer (in `TooltipController`) effets are triggered
3. New data attributes are set
4. Position calculation resolves, and position-related styles are set

Since 3 and 4 happen asynchronously, they resolve at different times (I believe 3 always resolves first), resulting in the tooltip rendering twice. 

`calculatingPosition` is used to hide the tooltip until 4 resolves, which ensures data attributes and the position have been updated.

This feels like a very hacky solution, so we should probably find a better way to make 3 and 4 happen on the same render cycle.